src/lib/y2partitioner/actions/move_partition.rb

```diff
@@ -0,0 +1,344 @@
+      #   adjacent free space.
+      def previous_adjacent_unused_slot
+        return nil unless previous_unused_slot
+        return previous_unused_slot unless previous_partition
+
+        previous_unused_slot.region.start > previous_partition.region.start ? previous_unused_slot : nil
+      end
+
+      # Next adjacent free space where the partition can be moved to
+      #
+      # @return [Y2Storage::PartitionTables::PartitionSlot, nil] nil if there is no next
+      #   adjacent free space.
+      def next_adjacent_unused_slot
+        return nil unless next_unused_slot
+        return next_unused_slot unless next_partition
+
+        next_unused_slot.region.start < next_partition.region.start ? next_unused_slot : nil
+      end
+
+      # Previous free space, not necessary adjacent to the partition
+      #
+      # @note In case the partition to move is a primary partition, free spaces inside an extended
+      #   partition are not taken into account. Otherwise, when the partition is logical, only free
+      #   spaces inside the extended partition are considered.
+      #
+      # @return [Y2Storage::PartitionTables::PartitionSlot, nil] nil if there is no previous free space.
+      def previous_unused_slot
+        unused_slots(partition.type).select { |s| s.region.start < partition.region.start }.last
+      end
+
+      # Next free space, not necessary adjacent to the partition
+      #
+      # @note In case the partition to move is a primary partition, free spaces inside an extended
+      #   partition are not taken into account. Otherwise, when the partition is logical, only free
+      #   spaces inside the extended partition are considered.
+      #
+      # @return [Y2Storage::PartitionTables::PartitionSlot, nil] nil if there is no next free space.
+      def next_unused_slot
+        unused_slots(partition.type).select { |s| s.region.start > partition.region.start }.first
+      end
+
+      # Partition placed before the partition to move
+      #
+      # @note In case the partition to move is a primary partition, logical partitions are not taken
+      #   into account. Otherwise, when the partition is logical, only other logical partitions are
+      #   considered.
+      #
+      # @return [Y2Storage::Partition, nil] nil if there is no partition before.
+      def previous_partition
+        partitions(partition.type).select { |p| p.region.start < partition.region.start }.last
+      end
+
+      # Partition placed after the partition to move
+      #
+      # @note In case the partition to move is a primary partition, logical partitions are not taken
+      #   into account. Otherwise, when the partition is logical, only other logical partitions are
+      #   considered.
+      #
+      # @return [Y2Storage::Partition, nil] nil if there is no partition after.
+      def next_partition
+        partitions(partition.type).select { |p| p.region.start > partition.region.start }.first
```

Comment:
same here. And btw maybe constant Proc can be used here for predicates, but just nice to have.